src/MusicStore/Program.cs

```diff
@@ -37,62 +39,96 @@ public static void Main(string[] args)
-            Console.WriteLine();
+            int serverStartupTime = (int)totalTime.ElapsedMilliseconds;
+            eventSource.ServerStartupEnd(serverStartupTime);
+
+
+
+
 
             using (var client = new HttpClient())
             {
-                Console.WriteLine("Starting request to http://localhost:5000");
                 var requestTime = Stopwatch.StartNew();
+                eventSource.FirstRequestBegin();
                 var response = client.GetAsync("http://localhost:5000").Result;
                 response.EnsureSuccessStatusCode(); // Crash immediately if something is broken
                 requestTime.Stop();
-                var firstRequestTime = requestTime.ElapsedMilliseconds;
+                int firstRequestTime = (int)requestTime.ElapsedMilliseconds;
+                eventSource.FirstRequestEnd(firstRequestTime);
 
-                Console.WriteLine("Response: {0}", response.StatusCode);
-                Console.WriteLine("Request took {0}ms", firstRequestTime);
+                Console.WriteLine("============= Startup Performance ============");
+                Console.WriteLine();
+                Console.WriteLine("Server start (ms): {0,5}", serverStartupTime);
+                Console.WriteLine("1st Request (ms):  {0,5}", firstRequestTime);
+                Console.WriteLine("Total (ms):        {0,5}", serverStartupTime + firstRequestTime);
                 Console.WriteLine();
-                Console.WriteLine("Cold start time (server start + first request time): {0}ms", serverStartupTime + firstRequestTime);
                 Console.WriteLine();
                 Console.WriteLine();
-                
-                var minRequestTime = long.MaxValue;
-                var maxRequestTime = long.MinValue;
-                var averageRequestTime = 0.0;
 
-                Console.WriteLine("Running 100 requests");
-                for (var i = 1; i <= 100; i++)
+                if (args.Length == 0 || args[0] != "-skipSteadyState")
                 {
-                    requestTime.Restart();
-                    response = client.GetAsync("http://localhost:5000").Result;
-                    requestTime.Stop();
-
-                    var requestTimeElapsed = requestTime.ElapsedMilliseconds;
-                    if (requestTimeElapsed < minRequestTime)
-                    {
-                        minRequestTime = requestTimeElapsed;
-                    }
-
-                    if (requestTimeElapsed > maxRequestTime)
+                    int[] threshholds = new int[] { 500, 1000, 1500, 2000, 10000 };
+                    double totalTimeMs = 0;
+                    int totalRequests = 0;
+                    Console.WriteLine("========== Steady State Performance ==========");
+                    Console.WriteLine();
+                    Console.WriteLine("  Requests    Aggregate Time(ms)    Req/s   Request Min(ms)   Request Max(ms)");
```

Comment:
Is aggregate time useful? It would be nice to have a standard error % column, maybe I can add it later.

Reply:
What sort of error metric are you looking for, std. deviation of the requests in that batch? difference between average of earlier batches and later ones?

For aggregate time I wanted to be able to make comparisons such as "Serving 1500 requests with tiered jitting enabled is faster than serving 1500 requests without it." It might be the case that requests 1001-1500 took longer with tiered jitting, but if startup gave us enough of a win then in aggregate tiered jitting might still be winning the race at that point.

Reply:
RE aggregate time, I see, that would be useful

For error I usually use the standard error of the mean (https://en.wikipedia.org/wiki/Standard_error) given as a percentage of the mean. That makes it easy to compare the error between batches and across runs on different changes.

Reply:
Probably would involve running multiple smaller batches per batch to create a sample set, and would give an idea about how much the value is varying within that batch.